src/org.hdfgroup.hdfview/hdf/view/TableView/DefaultScalarDSTableView.java

```diff
@@ -113,7 +113,12 @@ public DefaultScalarDSTableView(DataViewManager theView, HashMap dataPropertiesM
         super(theView, dataPropertiesMap);
 
         if (!shell.isDisposed()) {
-            shell.setImage(dataObject.getDatatype().isText() ? ViewProperties.getTextIcon() : ViewProperties.getDatasetIcon());
+            if (System.getProperty("os.name").toLowerCase().contains("mac")) {
+                shell.setImages(ViewProperties.getHdfIcons());
```

Comment:
I can see from your example screenshots that the dataset shell still uses the Dataset icon, but wouldn't this line explicitly set its icon to the HDF icon on macOS?

Reply:
@jhendersonHDF yes, it keeps dock icon as the HDF icon when you open a dataset (i.e. the second icon from the right). The icon on the right side is the original code using HDFView 3.1.4 downloaded from HDFGroup's website.

I just seems odd for an application to change the macOS dock icon.

Reply:
Ah ok. Generally on Windows and Linux each dataset would be opened in a separate window with its own Dataset icon, while the main HDFView window would keep the HDF icon; I guess that isn't the case for macOS? In any case I think this is a great improvement for users of HDFView on macOS.

Reply:
They still get open in a new window on macOS, but there is no icon in the corner of the window like there is in Windows and Linux.

<img width="792" alt="Screen Shot 2022-09-12 at 12 45 56 PM" src="https://user-images.githubusercontent.com/2364606/189712431-812f072b-bdc6-4f25-898f-ab15689d4f8c.png">